go.mod

```diff
@@ -26,4 +26,36 @@ replace github.com/btcsuite/btcd/btcutil => ./btcutil
 
 replace github.com/btcsuite/btcd/btcec/v2 => ./btcec
 
+// The retract statements below fixes an accidental push of the tags of a btcd
+// fork.
+retract (
+	v0.18.1
+	v0.18.0
+	v0.17.1
+	v0.17.0
+	v0.16.5
+	v0.16.4
+	v0.16.3
+	v0.16.2
+	v0.16.1
+	v0.16.0
+
+	v0.15.2
+	v0.15.1
+	v0.15.0
+
+	v0.14.7
+	v0.14.6
+	v0.14.6
+	v0.14.5
+	v0.14.4
+	v0.14.3
+	v0.14.2
+	v0.14.1
+
+	v0.14.0
+	v0.13.0-beta2
+	v0.13.0-beta
+)
```

Comment:
That appears to be all the foreign tags back to the fork point.

```
$ git log --tags --simplify-by-decoration --pretty="format:%d - %cr - %an - %cd"

(HEAD -> master, tag: btcec/v2.0.0, origin/master, origin/HEAD) - 9 days ago - Olaoluwa Osuntokun - Wed Jan 26 16:36:21 2022 -0800
 (tag: btcutil/v1.0.0, tag: btcutil/psbt/v1.0.0) - 4 weeks ago - Olaoluwa Osuntokun - Mon Jan 10 19:27:46 2022 -0800
 (computepks) - 3 months ago - naveen - Tue Oct 26 10:00:04 2021 -0400
 (chappjc/master, update-addr-copy) - 4 months ago - Olaoluwa Osuntokun - Tue Oct 5 11:44:31 2021 -0700
 (tag: v0.22.0-beta) - 8 months ago - Anirudha Bose - Tue Jun 1 13:16:51 2021 -0400
 (tag: v0.21.0-beta) - 1 year, 5 months ago - John C. Vernaleo - Fri Aug 28 08:13:36 2020 -0400
 (tag: v0.20.1-beta) - 2 years, 3 months ago - Olaoluwa Osuntokun - Tue Nov 12 18:27:49 2019 -0800
 (tag: v0.20.0-beta) - 2 years, 4 months ago - Olaoluwa Osuntokun - Tue Oct 15 10:11:03 2019 +0200
 (tag: v0.18.1) - 9 months ago - Josh Ellithorpe - Mon May 24 04:20:07 2021 -0700
 (tag: v0.18.0) - 9 months ago - quest - Wed May 12 15:46:14 2021 -1000
 (tag: v0.17.1) - 1 year, 2 months ago - quest - Tue Dec 8 16:02:04 2020 -1000
 (tag: v0.17.0) - 1 year, 3 months ago - Chris Pacia - Sat Oct 24 11:36:01 2020 -0400
 (tag: v0.16.5) - 1 year, 6 months ago - Josh Ellithorpe - Thu Jul 23 17:13:02 2020 -0700
 (tag: v0.16.4) - 1 year, 7 months ago - Josh Ellithorpe - Mon Jul 6 13:20:10 2020 -0700
 (tag: v0.16.3) - 1 year, 8 months ago - Chris - Wed Jun 10 18:57:39 2020 -0400
 (tag: v0.16.2) - 1 year, 9 months ago - Chris - Mon May 18 19:16:24 2020 -0400
 (tag: v0.16.1) - 1 year, 9 months ago - Chris - Fri May 15 13:04:05 2020 -0400
 (tag: v0.16.0) - 1 year, 9 months ago - Chris Pacia - Tue May 5 20:27:50 2020 -0400
 (tag: v0.15.2) - 2 years, 2 months ago - Chris - Sat Nov 30 22:08:23 2019 -0500
 (tag: v0.15.1) - 2 years, 3 months ago - Chris Pacia - Sun Nov 17 16:08:28 2019 -0500
 (tag: v0.15.0) - 2 years, 4 months ago - Chris Pacia - Sun Oct 20 20:05:24 2019 -0400
 (tag: v0.14.7) - 2 years, 6 months ago - Chris - Tue Aug 20 14:32:48 2019 -0400
 (tag: v0.14.6) - 2 years, 8 months ago - Chris Pacia - Sat Jun 15 21:42:38 2019 -0400
 (tag: v0.14.5) - 2 years, 8 months ago - Chris Pacia - Thu Jun 13 13:47:20 2019 -0400
 (tag: v0.14.4) - 2 years, 8 months ago - Chris Pacia - Sun May 26 09:54:39 2019 -0400
 (tag: v0.14.3) - 2 years, 9 months ago - Chris Pacia - Fri May 3 07:38:14 2019 -0400
 (tag: v0.14.2) - 2 years, 10 months ago - Chris Pacia - Wed Apr 17 15:12:04 2019 -0400
 (tag: v0.14.1) - 2 years, 10 months ago - Chris Pacia - Wed Apr 10 22:43:25 2019 -0400
 (tag: v0.14.0) - 2 years, 11 months ago - Josh Ellithorpe - Thu Mar 21 11:17:44 2019 -0700
 (tag: v0.13.0-beta2) - 3 years, 2 months ago - Chris Pacia - Tue Dec 18 22:42:32 2018 -0500
 (tag: v0.13.0-beta) - 3 years, 3 months ago - Chris Pacia - Sun Nov 18 10:01:16 2018 -0500
 - 4 years, 6 months ago - Dave Collins - Tue Aug 15 20:09:19 2017 -0500
 - 8 years ago - Dave Collins - Wed May 7 11:14:39 2014 -0500
 (tag: BTCD_0_12_0_BETA) - 6 years ago - Dave Collins - Fri Nov 20 19:33:17 2015 -0600
 (tag: BTCD_0_11_1_BETA) - 7 years ago - Dave Collins - Wed May 27 11:06:52 2015 -0500
 (tag: BTCD_0_11_0_BETA) - 7 years ago - Dave Collins - Wed May 6 09:58:25 2015 -0500
 (tag: BTCD_0_10_0_BETA) - 7 years ago - Dave Collins - Sun Mar 1 20:18:11 2015 -0600
```

I'm not 100% on how module retraction works though, but I think this will do it.

Reply:
Thanks for verifying!

On the `lnd` side, we've succesful used it to stop naive users from using `go get -v` to fetch some really old tags back before our actual production release. We have the same issue in that repo, when Go sees the `-beta` suffix, it considers any other tag w/o that, as being newer.

Reply:
After I made this PR, I also _deleted_ all the tags from Github's PoV, so a `git fetch` by new contributors doesn't taint their local cache.

Reply:
On my side, it seems to be working well again. No need to use the `replace` fix to make it work.
Thanks.

Reply:
> On my side, it seems to be working well again. No need to use the replace fix to make it work.
Nice! The bogus tags are still here: https://proxy.golang.org/github.com/btcsuite/btcd/@v/list 

So I think we may need to keep the replace directives after all.

Reply:
It sometimes takes a while.  There's an update interval, something like 30 minutes for the module proxy.  Eventually the following command should stop showing the tags: `go list -m -versions github.com/btcsuite/btcd`
unless the `-retracted` flag is also provided.